### ISSUE TITLE
ipsec auto --up on connection that is up should trigger ipsec-rekey

### ISSUE THREAD

**Issue by letoams** [libreswan/libreswan#366]
ipsec auto --up on connection that is up should trigger ipsec-rekey

For some details on this, see ikev2-cp-rekey-01

**Comment by cagney** [comment]
Does this still apply?  The test uses ipsec-rekey vis:
```
 ipsec whack --rekey-ipsec --name eastnet-any
002 "eastnet-any"[1] 192.1.2.23 #3: initiating rekey to replace Child SA #2
1v2 "eastnet-any"[1] 192.1.2.23 #3: sent CREATE_CHILD_SA request to rekey IPsec SA
004 "eastnet-any"[1] 192.1.2.23 #3: initiator rekeyed Child SA #2 using #1; IPsec tunnel [100.64.13.2-100.64.13.2:0-65535 0] -> [192.0.2.0-192.0.2.255:0-65535 0] {ESP/ESN=>0xESPESP <0xESPESP xfrm=AES_GCM_16_256-NONE-MODP2048 DPD=passive}
002 "eastnet-any"[1] 192.1.2.23 #2: deleting established Child SA using IKE SA #1
005 "eastnet-any"[1] 192.1.2.23 #2: ESP traffic information: in=84B out=84B
```

**Comment by letoams** [comment]
On Wed, 21 Jun 2023, cagney wrote:

> Does this still apply? The test uses ipsec-rekey vis:
>
>  ipsec whack --rekey-ipsec --name eastnet-any
> 002 "eastnet-any"[1] 192.1.2.23 #3: initiating rekey to replace Child SA #2
> 1v2 "eastnet-any"[1] 192.1.2.23 #3: sent CREATE_CHILD_SA request to rekey IPsec SA
> 004 "eastnet-any"[1] 192.1.2.23 #3: initiator rekeyed Child SA #2 using #1; IPsec tunnel [100.64.
> 13.2-100.64.13.2:0-65535 0] -> [192.0.2.0-192.0.2.255:0-65535 0] {ESP/ESN=>0xESPESP <0xESPESP xfr
> m=AES_GCM_16_256-NONE-MODP2048 DPD=passive}
> 002 "eastnet-any"[1] 192.1.2.23 #2: deleting established Child SA using IKE SA #1
> 005 "eastnet-any"[1] 192.1.2.23 #2: ESP traffic information: in=84B out=84B

It used to be that running --up again was the only way to trigger a
rekey. I think now it would be more correct to error with "already up"
and keep rekey to its seperate command.

Maybe even "connection is already up - did you mean --rekey-ike or
--rekey-ipsec".

I think currently we have no "ipsec" wrapper for rekey, only whack commands?
eg, next to "ipsec up" we should have "ipsec rekey" and "ipsec rekey-ike" ?

Paul

**Comment by cagney** [comment]
Looks like the test is `ikev2-cp-rekey-02-up-up`

**Comment by cagney** [comment]
> I think currently we have no "ipsec" wrapper for rekey, only whack commands? eg, next to "ipsec up" we should have "ipsec rekey" and "ipsec rekey-ike" ?

yes agreed; although `ipsec rekey` is IKEv2 only; I wonder what injecting `EVENT_v1_REPLACE` will do ...

**Comment by cagney** [comment]
Per 9ea7cbf0f438f4782daedb3f188db11b169d9c52 it is now clear it is doing a replace:
```
--- a/testing/pluto/ikev2-13-ah/west.console.txt
+++ b/testing/pluto/ikev2-13-ah/west.console.txt
@@ -46,6 +46,7 @@ west #
 west #
  ipsec auto --up westnet-eastnet-ipv4-psk-ikev2
 002 "westnet-eastnet-ipv4-psk-ikev2" #3: initiating Child SA using IKE SA #1
+002 "westnet-eastnet-ipv4-psk-ikev2" #3: connection is already erouted
 1v2 "westnet-eastnet-ipv4-psk-ikev2" #3: sent CREATE_CHILD_SA request for new IPsec SA
 004 "westnet-eastnet-ipv4-psk-ikev2" #3: initiator established Child SA using #1; IPsec tunnel [192.0.1.0-192.0.1.255:0-65535 0] -> [192.0.2.0-192.0.2.255:0-65535 0] {AH/ESN=>0xAHAH <0xAHAH xfrm=HMAC_SHA1_96 DPD=passive}
 west #
```
